packages/wallet/WalletManager.js

```diff
@@ -0,0 +1,421 @@
+   */
+  setSeed() {
+    if (arguments.length === 0) {
+      // generate seed; both network uses the same bip39 module to generate seed phrases
+      switch (this.network) {
+        case 'stellar':
+        case 'ethereum':
+          this.seed = bip39.generateMnemonic();
+          // default strength = 128, 12 words
+          // 160 - 18 words; 224 - 21 words; 256 - 24 words
+          break;
+        default:
+          this.seed = null;
+          console.log('The network has not been set.');
+      }
+    } else if (arguments.length === 1) {
+      if ([12, 18, 21, 24].includes(arguments[0])) {
+        const strength = (function toBits(numberOfWords) {
+          switch (numberOfWords) {
+            case 12:
+              return 128;
+            case 18:
+              return 160;
+            case 21:
+              return 224;
+            case 24:
+              return 256;
+            default:
+              return 128;
+          }
+        }(arguments[0]));
+        this.seed = bip39.generateMnemonic(strength);
+      } else if (typeof (arguments[0]) === 'string') {
+        if (bip39.validateMnemonic(arguments[0])) {
+          this.seed = arguments[0];
+        } else {
+          this.seed = null;
+          console.log('The input is not a set of valid seed phrases.');
+        }
+      } else {
+        this.seed = null;
+        console.log('The number of seed phrases must be one of 12, 18, 21, 24.');
+      }
+    } else {
+      console.log('The argument must be empty, number of seed phrases, or valid seed phrases.');
+    }
+    return this.seed;
+  }
+
+  /**
+   * @return {string} the current seed
+   */
+  getSeed() {
+    if (this.seed == null) {
+      console.log('The seed of the wallet is not set.');
+      return null;
+    }
+    return this.seed;
+  }
+
+  /**
+   * Generate the wallet based on the seed
+   */
+  setWallet() {
+    switch (this.network) {
+      case 'stellar':
+        this.wallet = stellarHDWallet.fromMnemonic(this.seed);
+        break;
+      case 'ethereum':
+        this.wallet = hdkey.fromMasterSeed(this.seed);
+        break;
+      default:
+        this.wallet = null;
+        console.log('The seed is not specified or not valid.');
+    }
+    return this.wallet;
+  }
+
+  /**
+   * @returns {Object} the current wallet
+   */
+  getWallet() {
+    if (this.wallet == null) {
+      console.log('The wallet is not set.');
+      return null;
+    }
+    return this.wallet;
+  }
+
+  /**
+   * Before generating accounts, the seed and wallet MUST be there.
+   * i.e., random account generatation is NOT allowed.
+   * If the wallet is set, generate the first (0) account in the wallet;
+   * If the parameter is a number,
+   * Generate the account in the wallet. Its index is the number.
+   * If the parameter is a string,
+   * Use it as the private/public key to generate the account
+   * @returns {Account} the account created/imported
+   */
+  async getAccount() {
```

Comment:
Function header should mention that the accounts get pushed into account array.

On that point, what is the purpose of `accountArray`? It looks like it stores the history of account generated and it allows for repeat of accounts in the array.